pkg/virt-launcher/virtwrap/api/schema_test.go

```diff
@@ -230,8 +304,13 @@ var _ = Describe("Schema", func() {
 	})
 	Context("With example schema", func() {
 		It("Unmarshal into struct", func() {
+			var err error
 			newDomain := DomainSpec{}
-			err := xml.Unmarshal([]byte(exampleXML), &newDomain)
+			if runtime.GOARCH == "ppc64le" {
```

Comment:
In a follow-up we should just execute all unit tests for all platforms always and not rely on GOARCH.

Reply:
I might be missing your point, but aside from the privilege tests that the patch made skip in Power, the other ones just had their expected output adjusted to the architecture. I don't see a way to avoid the GOARCH in some cases given for example that in Power we don't support machine type q35 and use pseries, which in turn I assume isn't a recognized machine type for amd64 as well. So these kind of subtleties need to be taken into account occasionally depending on the test.

Reply:
There may be unit tests which depend on some arch specific features but that should be rare. In this case, all we need to do, is pull the GOARCH check out to the outmost layer (or something like that), and pass in the architecture as a parameter/member.

Then we can  inject the architecture under test into the code under test and the unit tests.

That allows running all unit test locally and iterate faster over the code ...